_posts/2020-09-15-data-sharing-at-scale.md

```diff
@@ -0,0 +1,301 @@
+      DatabaseName: !Ref MySourceGlueDatabase
+      Targets:
+        S3Targets:
+          - Path: !Ref MySourceDataStore
+      SchemaChangePolicy:
+        UpdateBehavior: 'UPDATE_IN_DATABASE'
+        DeleteBehavior: 'LOG'
+
+  SourceCrawlerLakeGrants:
+    Type: AWS::LakeFormation::Permissions
+    Properties:
+      DataLakePrincipal:
+        DataLakePrincipalIdentifier: !GetAtt MySourceCrawlerRole.Arn
+      Permissions:
+        - ALTER
+        - DROP
+        - CREATE_TABLE
+      Resource:
+        DatabaseResource:
+          Name: !Ref MySourceGlueDatabase
+
+  DatalakeLocation:
+    Type: AWS::LakeFormation::Resource
+    Properties: 
+      ResourceArn: !GetAtt  MySourceDataStore.Arn
+      RoleArn: !Sub arn:aws:iam::${AWS::AccountId}:role/aws-service-role/lakeformation.amazonaws.com/AWSServiceRoleForLakeFormationDataAccess
+      UseServiceLinkedRole: true
+```
+
+Add the above resources to your datalake.yml file and run an update stack command.
+
+```sh
+aws cloudformation update-stack --stack-name my-source-glue-stack --template-body file://datalake.yml --capabilities "CAPABILITY_NAMED_IAM"
+```
+
+When you have deployed the crawler, you should be able to see and run it in the Glue Console. 
+
+<center><img src="/img/lake-formation/04.png" /></center><br/>
+
+Click on run and wait for your crawler to be completed.
+
+Once your crawler runs successfully, you can go to tables menu item and see your table being generated.
+
+<center><img src="/img/lake-formation/05.png" /></center><br/>
+
+## Cross-account grant
+
+Once you have your Glue database and tables created, you will be able to manage the database access via Lake Formation. 
+
+To enable cross-account access, you will need to add a Lake Formation grant and specify the consumer account number.
+
+```yml
+ CrossAccountLakeGrants:
+    Type: AWS::LakeFormation::Permissions
+    Properties:
+      DataLakePrincipal:
+        DataLakePrincipalIdentifier: "300000000015" // Consumer account number
+      Permissions:
+        - SELECT
+      PermissionsWithGrantOption:
+        - SELECT
+      Resource:
+        TableResource:
+          DatabaseName: !Ref MySourceGlueDatabase
+          Name: !Sub 'my_source_data_store_ap_southeast_2_${AWS::AccountId}'
+          # TableWildcard: [] // WhildCard is not available via cloudformation yet 
+```
+
+This will automatically create a resource share with AWS Resource Access Manager (RAM), and the administrator will see the shared database in their Glue Catalog.
+
+## Setting up permissions in the consumer account
+
+We have shared the table from the source account to the consumer account. But, by default only the administrator of the consumer lake will have access to those shared resources. To enable other IAM users or roles to access the resources, we'll need to create grants.
+
+Login into the consumer account and setup Lake Formation base settings:
+
+1) Setup a Lake Formation administrator:
```

Comment:
might be worth using a sub-heading for these? Since you're at the h2 level already, suggest ###
eg, 
```
### 1) Setup a Lake Formation administrator:
```